src/Services/cdl/Debugger.js

```
@@ -108,53 +92,129 @@ class Debugger {
+        do {
+            position = this.cdl._getPreviousPosition(position);
 
-        while (--position >= 0) {
-            const currStack = this.cdl.getCallStackAtPosition(position);
-            if (stack.length >= currStack.length) {
-                this.getPositionData(position);
+            if (position == null) {
                 return;
             }
+
+            for (let breakpoint of this.breakpoints) {
```

ESLint

[error] 'breakpoint' is never reassigned. Use 'const' instead.

(prefer-const)